core-tests/shared/src/test/scala/zio/stm/TQueueSpec.scala

```diff
@@ -0,0 +1,194 @@
+          ans  <- tq.takeUpTo(3)
+          size <- tq.size
+        } yield (ans, size)
+        for {
+          z <- t.commit
+        } yield assert(z._2)(equalTo(2)) &&
+          assert(z._1)(hasSameElements(List(1, 2, 3)))
+      },
+      testM("offerAll & takeAll") {
+        val t = for {
+          tq  <- TQueue.make[Int](5)
+          _   <- tq.offerAll(List(1, 2, 3, 4, 5))
+          ans <- tq.takeAll
+        } yield ans
+        assertM(t.commit)(hasSameElements(List(1, 2, 3, 4, 5)))
+      },
+      testM("takeUpTo") {
+        val t = for {
+          tq   <- TQueue.make[Int](5)
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          ans  <- tq.takeUpTo(3)
+          size <- tq.size
+        } yield (ans, size)
+        for {
+          z <- t.commit
+        } yield assert(z._2)(equalTo(2)) &&
+          assert(z._1)(hasSameElements(List(1, 2, 3)))
+      },
+      testM("takeUpTo larger than container") {
+        val t = for {
+          tq   <- TQueue.make[Int](5)
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          ans  <- tq.takeUpTo(7)
+          size <- tq.size
+        } yield (ans, size)
+        for {
+          z <- t.commit
+        } yield assert(z._2)(equalTo(0)) &&
+          assert(z._1)(hasSameElements(List(1, 2, 3, 4, 5)))
+      },
+      testM("poll value") {
+        val t = for {
+          tq  <- TQueue.make[Int](5)
+          _   <- tq.offerAll(List(1, 2, 3))
+          ans <- tq.poll
+        } yield ans
+        assertM(t.commit)(isSome(equalTo(1)))
+      },
+      testM("poll empty queue") {
+        val t = for {
+          tq  <- TQueue.make[Int](5)
+          ans <- tq.poll
+        } yield ans
+        assertM(t.commit)(isNone)
+      },
+      testM("seek element") {
+        val t = for {
+          tq   <- TQueue.make[Int](5)
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          ans  <- tq.seek(_ == 3)
+          size <- tq.size
+        } yield (ans, size)
+        for {
+          z <- t.commit
+        } yield assert(z._1)(equalTo(3)) &&
+          assert(z._2)(equalTo(2))
+      }
+    ),
+    suite("lookup")(
+      testM("size") {
+        val t = for {
+          tq   <- TQueue.unbounded[Int]
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          size <- tq.size
+        } yield size
+        assertM(t.commit)(equalTo(5))
+      },
+      testM("peek the next value") {
+        val t = for {
+          tq   <- TQueue.unbounded[Int]
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          next <- tq.peek
+          size <- tq.size
+        } yield (next, size)
+        for {
+          z <- t.commit
+        } yield assert(z._1)(equalTo(1)) &&
+          assert(z._2)(equalTo(5))
+      },
+      testM("view the last value") {
+        val t = for {
+          tq   <- TQueue.unbounded[Int]
+          _    <- tq.offerAll(List(1, 2, 3, 4, 5))
+          last <- tq.last
+          size <- tq.size
+        } yield (last, size)
+        for {
```

Comment:
It might be slightly more readable in the following form:

```scala
assertM(t.commit)(equalTo((5,5)))
```

Also, one nitpick, just to align with other tests prefer `tx` to `t`.